### PR TITLE
Parse signatures and handle enum etc better

### PR DESCRIPTION
**Pull request by liach**

### CODE REVIEW

**Comment by liach** [comment]
Should I do type variables after this pr is merged?

File: build.gradle [hunk 1]
```diff
@@ -18,11 +19,20 @@ repositories {
...
+
+    testImplementation 'org.junit.jupiter:junit-jupiter-api:5.1.0'
+    testRuntimeOnly 'org.junit.jupiter:junit-jupiter-engine:5.1.0'
+}
+
+test {
+    useJUnitPlatform()
 }
+
+tasks.assemble.dependsOn shadowJar
```
Comment (modmuss50) [inline build.gradle]: is the shadow jar needed? shouldnt the deps be pulled in via maven using it in yarn?

Comment (liach) [reply build.gradle]: I just use the shadow jar so that grabbing it into a local directory is way easier.

**Comment by modmuss50** [comment]
Merging this to move to a new repo

**Pull request merged**